astrobee/survey_manager/survey_planner/src/isaac_action_node.cpp

```diff
@@ -44,62 +159,70 @@ IsaacAction::IsaacAction(ros::NodeHandle nh, const std::string& action, const st
 void IsaacAction::do_work() {
   std::string from, towards;
 
-  const std::vector<std::string>& command_args = get_arguments();
-  if (command_args.size() < 3) {
-    finish(false, 1.0, "Not enough arguments for [MOVE] command");
-  }
-
   // Start process if not started yet
   if (progress_ == 0.0) {
+    const std::vector<std::string>& command_args = get_arguments();
+    if (command_args.size() < 3) {
+      finish(false, 1.0, "Not enough arguments for [MOVE] command");
+    }
+
+    // Get estimated action duration
+    action_duration_ = get_action_duration(action_name_);
```

Comment:
Would it be possible to do this in the constructor, since (if I understand correctly?) this will get called again for each time we do the action vs. only once in the constructor?

Reply:
Good thought. (It takes < 1 second, so not very critical.)

Reply:
Fixed in https://github.com/marinagmoreira/isaac/pull/2/commits/f49dd6cdc257bf30160cfeecccc0fef6b50c8900